components/contract-components/hooks.ts

```diff
@@ -531,13 +536,31 @@ export function useCustomContractDeployMutation(
         }
 
         // deploy contract
-        contractAddress = await sdk.deployer.deployContractFromUri(
-          ipfsHash.startsWith("ipfs://") ? ipfsHash : `ipfs://${ipfsHash}`,
-          Object.values(data.deployParams),
-          {
-            forceDirectDeploy,
-          },
-        );
+        // Handle ZkSync deployments separately
+        const isZkSync =
+          chainId === ZksyncEraTestnet.chainId || chainId === ZksyncEra.chainId;
+        if (isZkSync) {
+          // Get metamask signer using zksync-web3 library -- for custom fields in signature
+          const zkSigner = new Web3Provider(
+            window.ethereum as unknown as providers.ExternalProvider,
```

Comment:
This assumes that user is using injected wallet, but that is not always true right?